core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -64,6 +65,19 @@ public function getConfigTreeBuilder(): TreeBuilder
                 ->end()
                 ->variableNode('localconfig')
                     ->info('Allows to set TL_CONFIG variables, overriding settings stored in localconfig.php. Changes in the Contao back end will not have any effect.')
+                    ->validate()
+                        ->always(
+                            static function (array $options): array {
+                                foreach (array_keys($options) as $option) {
+                                    if ($newKey = Config::getNewKey($option)) {
+                                        trigger_deprecation('contao/core-bundle', '4.12', 'Setting "contao.localconfig.%s" has been deprecated. Use "%s" instead.', $option, $newKey);
```

Comment:
We could even set the new key here instead of the old one.

Reply:
This would not work (easily) because `Config::getNewKey()` also returns non-settable keys (from `$arrDeprecated`).

Reply:
True. 🦊